finagle-http/src/main/scala/com/twitter/finagle/http/codec/HttpServerDispatcher.scala

```diff
@@ -31,6 +31,11 @@ private[finagle] class HttpServerDispatcher(
     service.close()
   }
 
+  private def mayHaveContent(status: Status): Boolean = status match {
```

Comment:
TLDR: I've had some more time to think about it, and I'm not certain of an easy solution.

> Umm.. I think Content-Length header field is not set to 0 for a 204 response when there's no message-body. (But I think it's correct that the header field is added when we explicitly send a message-body, which is our problem here.)
> More exactly, in the case of a 204 response with no body, rep.contentLength is None and rep.content.length is 0 I suppose.

Sorry, its not clear to me if we agree. I believe at a `Response` with an empty body will get assigned a content length header of length 0. I haven't added tags to the `handle` function to check, but a raw `Response` should not be chunked and will not have a defined `contentLength`, so line 97 (on _master_, maybe this is the source of our confusion) should set the content-length to 0 on a 204 response without a body.

I don't think this can be solved _only_ in the compliance `Filter`, there needs to be some coupling because, as is, the content-length is going to be set for non-chunked bodies regardless of status code. I personally would hate to make the `HttpServerDispatcher` aware of a single `Filter`, that just doesn't seem fair to the other filters. 😄  That suggests to me (and we should get @mosesn's opinion in light of what we've found) that this isn't actually a job for a filter. This feels more like the things that are getting delegated to the netty pipeline such as [`RespondToExpectContinue`](https://github.com/twitter/finagle/blob/develop/finagle-http/src/main/scala/com/twitter/finagle/http/Codec.scala#L307).